src/Microsoft.AspNet.SignalR.JS/jquery.signalR.core.js

```diff
@@ -689,69 +695,112 @@
                         return;
                     }
 
-                    keepAliveData = connection._.keepAliveData;
-                    connection.appRelativeUrl = res.Url;
-                    connection.id = res.ConnectionId;
-                    connection.token = res.ConnectionToken;
-                    connection.webSocketServerUrl = res.WebSocketServerUrl;
+                    // redirect case
+                    if (res.AccessToken && res.Url) {
```

Comment:
We'll need to set a new `ProtocolVersion` value as part of this. If an older client receives this message, it will either try to connect directly to the app (ignoring the `Url` field) or produce an unhelpful error message.

My suggestion would be to change this to `if(res.ProtocolVersion === "2.0")` and ensure the SDK adds that value to the response.

Reply:
https://github.com/SignalR/SignalR/blob/master/src/Microsoft.AspNet.SignalR.Core/Infrastructure/ProtocolResolver.cs should also be updated to support higher version

Reply:
change to use `RedirectUrl` to distinguish with already in-use `Url`